k2/python/tests/linear_fsa_test.py

```diff
@@ -35,32 +42,40 @@ def test_fsa_vec(self):
             [8, 7, 9],
         ]
         num_symbols = sum([len(s) for s in symbols])
-        fsa = k2.linear_fsa(symbols)
-        assert len(fsa.shape) == 3
-        assert fsa.shape[0] == 3, 'There should be 3 FSAs'
-        expected_arcs = [
-            # fsa 0
-            [0, 1, 1],
-            [1, 2, 3],
-            [2, 3, 5],
-            [3, 4, -1],
-            # fsa 1
-            [0, 1, 2],
-            [1, 2, 6],
-            [2, 3, -1],
-            # fsa 2
-            [0, 1, 8],
-            [1, 2, 7],
-            [2, 3, 9],
-            [3, 4, -1]
-        ]
-        print(fsa.arcs.values()[:, :-1])
-        assert torch.allclose(
-            fsa.arcs.values()[:, :-1],  # skip the last field `scores`
-            torch.tensor(expected_arcs, dtype=torch.int32))
-        assert torch.allclose(
-            fsa.scores,
-            torch.zeros(num_symbols + len(symbols), dtype=torch.float32))
+
+        devices = [torch.device('cpu')]
+        if torch.cuda.is_available():
+            devices.append(torch.device('cuda', 0))
+
+        for device in devices:
+            fsa = k2.linear_fsa(symbols, device)
```

Comment:
@pzelasko
It is possible to convert a list of list of symbols to LinearFsa**s** on GPU,
which I think is faster than creating them one by one on CPU.

Reply:
got it - so you're suggesting to convert the list of strings into an FsaVec, and then iterate over the FsaVec to cache each Fsa independently?

Reply:
I don't know what role LRU cache plays here. At least, the graphs can be created and cached on GPU.

Not sure whether `k2.linear_fsa` is faster than `k2.create_fsa_vec`. If `k2.linear_fsa` is faster,
we may not need the cache.